src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -212,14 +218,10 @@ public Void run() {
      * @return the array of class and interface names; or null if classes is
      * null or empty
      */
-    private static String[] typeNames(List<Class<?>> classes) {
-        if (classes == null || classes.size() == 0)
-            return null;
-        int size = classes.size();
-        String[] ifaces = new String[size];
-        for (int i = 0; i < size; i++)
-            ifaces[i] = dotToSlash(classes.get(i).getName());
-        return ifaces;
+    private static List<ClassDesc> typeNames(List<Class<?>> classes) {
```

Comment:
I think this method is not strictly necessary. 

`clb.withInterfaceSymbols(typeNames(interfaces))` can simply be replaced with 

```
clb.withInterfaceSymbols(interfaces.stream().map(ProxyGenerator::toClassDesc).toList());
```

`interfaces` should be non-null.  We can add `Objects.requireNonNull(interfaces)` in static `generateProxyClass` method.